satpy/tests/reader_tests/test_viirs_sdr.py

```diff
@@ -471,12 +481,35 @@ def test_load_all_m_bts(self):
                      ])
         self.assertEqual(len(ds), 5)
         for d in ds.values():
+            self.assertTrue(np.issubdtype(d.dtype, np.float32))
             self.assertEqual(d.attrs['calibration'], 'brightness_temperature')
             self.assertEqual(d.attrs['units'], 'K')
             self.assertEqual(d.attrs['rows_per_scan'], 16)
             self.assertIn('area', d.attrs)
             self.assertIsNotNone(d.attrs['area'])
 
+    def test_load_dnb_sza_no_factors(self):
+        """Load DNB solar zenith angle with no scaling factors.
+
+        The angles in VIIRS SDRs should never have scaling factors so we test
+        it that way.
+
+        """
+        from satpy.readers import load_reader
+        r = load_reader(self.reader_configs)
+        loadables = r.select_files_from_pathnames([
+            'GDNBO_npp_d20120225_t1801245_e1802487_b01708_c20120226002130255476_noaa_ops.h5',
+        ])
+        r.create_filehandlers(loadables, {'include_factors': False})
+        ds = r.load(['dnb_solar_zenith_angle'])
+        self.assertEqual(len(ds), 1)
+        for d in ds.values():
+            self.assertTrue(np.issubdtype(d.dtype, np.float32))
+            self.assertEqual(d.attrs['units'], 'degrees')
+            self.assertEqual(d.attrs['rows_per_scan'], 16)
+            self.assertIn('area', d.attrs)
+            self.assertIsNotNone(d.attrs['area'])
```

Comment:
This block and the others like it can be refactored into a handfull of methods, like `_assert_16_rows_of_float_angles_with_area` or `_assert_32_rows_of_float_reflectance_without_area`

Reply:
Fine. I have trouble refactoring during bug fix PRs beyond what is needed to fix the bug. I'll try to get to it later today.

Reply:
Also notice the codebeat issue. In `adjust_scaling_factors`, the unit-based scaling could be refactored into a new method, thus reducing the complexity of `adjust_scaling_factors`.